src/sqlfluff/core/parser/grammar/anyof.py

```diff
@@ -121,7 +122,7 @@ def _prune_options(
 
     def _match_once(
         self, segments: Tuple[BaseSegment, ...], parse_context: ParseContext
-    ) -> MatchResult:
+    ) -> Tuple[MatchResult, Optional["MatchableType"]]:
```

Comment:
Very minor suggestion: Think it'd be having this return a `NamedTuple` instead, for readability?

Reply:
Personally I'm fine with it being a tuple here, as it's essentially just the same return type as `BaseGrammar._longest_trimmed_match` so we should probably change both if we were to change this.

IMO given that these are both private methods `NamedTuple` might serve to obfuscate the typing more than it helps?

Reply:
No worries. Two return values is not that many. When it reaches 3 or more, that's usually when I _really_ look for alternatives. 👍